src/commands/install-deps.yml

```diff
@@ -9,34 +9,32 @@ parameters:
     type: string
     default: "gems-v1"
   path:
-    description: "Installation path."
+    description: "Project path."
     default: "."
     type: string
-  bundler-version:
-    description: Configure which version of bundler to install and utilize.
-    default: 2.1.4
-    type: string
 steps:
   - when:
       condition: <<parameters.with-cache>>
       steps:
         - restore_cache:
             keys:
               - << parameters.key >>-{{ checksum "<< parameters.path >>/Gemfile.lock"  }}-{{ .Branch }}
+              - << parameters.key >>-{{ checksum "<< parameters.path >>/Gemfile.lock"  }}-
   - run:
       name: Bundle Install <<^parameters.with-cache>>(No Cache)<</parameters.with-cache>>
       command: |
-        if ! echo $(bundle version)  | grep -q <<parameters.bundler-version>>; then
-          echo "Installing bundler <<parameters.bundler-version>>"
-          gem install bundler:<<parameters.bundler-version>>
+        APP_BUNDLER_VERSION=$(cat Gemfile.lock | tail -1 | tr -d " ")
+        if ! echo $(bundle version)  | grep -q $APP_BUNDLER_VERSION; then
+          echo "Installing bundler $APP_BUNDLER_VERSION"
+          gem install bundler:$APP_BUNDLER_VERSION
         else
-          echo "bundler <<parameters.bundler-version>> is already installed."
+          echo "bundler $APP_BUNDLER_VERSION is already installed."
         fi
-        bundle install
+        bundle check <<parameters.path>>/vendor/bundle || bundle install --deployment
   - when:
       condition: <<parameters.with-cache>>
       steps:
         - save_cache:
             key: << parameters.key >>-{{ checksum "<< parameters.path >>/Gemfile.lock"  }}-{{ .Branch }}
             paths:
-              - << parameters.path >>
+              - << parameters.path >>/vendor/bundle
```

Comment:
@teka23 Hello, why don't you add `/vendor/bundle` into the `parameters.path`?

Reply:
https://github.com/CircleCI-Public/ruby-orb/blob/master/src/commands/install-deps.yml#L25-L26
https://github.com/CircleCI-Public/ruby-orb/blob/master/src/commands/install-deps.yml#L41

the `path` parameter is being used as a path to the application. That's why I am intentionally adding `vendor/bundle` here. 

@KyleTryon btw, what is the purpose of the `path` parameter? Is it for specifying bundle path? or is it a path to the application? (Some applications share the model, and they are often placed in the same repository. So, it is necessary to specify a path to the application. That's why I am asking.)

Running bundle with `--deployment` flag will always install gems in `/vendor/bundle ` 
It checks inconsistency between Gemfile and Gemfile.lock so it is necessary for proper testing.
https://github.com/CircleCI-Public/ruby-orb/pull/29/files#diff-e3934e757062e4935271387f0cf57773R48

Reply:
@KyleTryon 
I have decided to go with, @ganezasan 's suggestion and left path parameters functionality as-is.